src/network/networkprotocol.cpp

```diff
@@ -0,0 +1,229 @@
+		TOCLIENT_SPAWN_PARTICLE
+		TOCLIENT_ADD_PARTICLESPAWNER
+		TOCLIENT_DELETE_PARTICLESPAWNER
+	PROTOCOL_VERSION 18:
+		damageGroups added to ToolCapabilities
+		sound_place added to ItemDefinition
+	PROTOCOL_VERSION 19:
+		AO_CMD_SET_PHYSICS_OVERRIDE
+	PROTOCOL_VERSION 20:
+		TOCLIENT_HUDADD
+		TOCLIENT_HUDRM
+		TOCLIENT_HUDCHANGE
+		TOCLIENT_HUD_SET_FLAGS
+	PROTOCOL_VERSION 21:
+		TOCLIENT_BREATH
+		TOSERVER_BREATH
+		range added to ItemDefinition
+		drowning, leveled and liquid_range added to ContentFeatures
+		stepheight and collideWithObjects added to object properties
+		version, heat and humidity transfer in MapBock
+		automatic_face_movement_dir and automatic_face_movement_dir_offset
+			added to object properties
+	PROTOCOL_VERSION 22:
+		add swap_node
+	PROTOCOL_VERSION 23:
+		Obsolete TOSERVER_RECEIVED_MEDIA
+		Server: Stop using TOSERVER_CLIENT_READY
+	PROTOCOL_VERSION 24:
+		ContentFeatures version 7
+		ContentFeatures: change number of special tiles to 6 (CF_SPECIAL_COUNT)
+	PROTOCOL_VERSION 25:
+		Rename TOCLIENT_ACCESS_DENIED to TOCLIENT_ACCESS_DENIED_LEGAGY
+		Rename TOCLIENT_DELETE_PARTICLESPAWNER to
+			TOCLIENT_DELETE_PARTICLESPAWNER_LEGACY
+		Rename TOSERVER_PASSWORD to TOSERVER_PASSWORD_LEGACY
+		Rename TOSERVER_INIT to TOSERVER_INIT_LEGACY
+		Rename TOCLIENT_INIT to TOCLIENT_INIT_LEGACY
+		Add TOCLIENT_ACCESS_DENIED new opcode (0x0A), using error codes
+			for standard error, keeping customisation possible. This
+			permit translation
+		Add TOCLIENT_DELETE_PARTICLESPAWNER (0x53), fixing the u16 read and
+			reading u32
+		Add new opcode TOSERVER_INIT for client presentation to server
+		Add new opcodes TOSERVER_FIRST_SRP, TOSERVER_SRP_BYTES_A,
+			TOSERVER_SRP_BYTES_M, TOCLIENT_SRP_BYTES_S_B
+			for the three supported auth mechanisms around srp
+		Add new opcodes TOCLIENT_ACCEPT_SUDO_MODE and TOCLIENT_DENY_SUDO_MODE
+			for sudo mode handling (auth mech generic way of changing password).
+		Add TOCLIENT_HELLO for presenting server to client after client
+			presentation
+		Add TOCLIENT_AUTH_ACCEPT to accept connection from client
+		Rename GENERIC_CMD_SET_ATTACHMENT to AO_CMD_ATTACH_TO
+	PROTOCOL_VERSION 26:
+		Add TileDef tileable_horizontal, tileable_vertical flags
+	PROTOCOL_VERSION 27:
+		backface_culling: backwards compatibility for playing with
+		newer client on pre-27 servers.
+		Add nodedef v3 - connected nodeboxes
+	PROTOCOL_VERSION 28:
+		CPT2_MESHOPTIONS
+	PROTOCOL_VERSION 29:
+		Server doesn't accept TOSERVER_BREATH anymore
+		serialization of TileAnimation params changed
+		TAT_SHEET_2D
+		Removed client-sided chat perdiction
+	PROTOCOL VERSION 30:
+		New ContentFeatures serialization version
+		Add node and tile color and palette
+		Fix plantlike visual_scale being applied squared and add compatibility
+			with pre-30 clients by sending sqrt(visual_scale)
+	PROTOCOL VERSION 31:
+		Add tile overlay
+		Stop sending TOSERVER_CLIENT_READY
+	PROTOCOL VERSION 32:
+		Add fading sounds
+	PROTOCOL VERSION 33:
```

Comment:
do we want to cut the list off here because everything above is 0.4.x or older?

Reply:
Yes